### PR TITLE
Add 80ch to first p of LegacyCard section

### PR DESCRIPTION
**Pull request by tjonx**
### WHY are these changes introduced?

Fixes #0000 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/LegacyCard/LegacyCard.scss [hunk 1]
```diff
@@ -59,6 +59,11 @@
         border-bottom-left-radius: var(--p-border-radius-3);
         border-bottom-right-radius: var(--p-border-radius-3);
       }
+
+      // stylelint-disable-next-line -- se23 override
+      .Section p:first-of-type {
+        max-width: 80ch;
+      }
```
Comment (sophschneider) [inline polaris-react/src/components/LegacyCard/LegacyCard.scss:66]: I don't think this selector will work well as a lot of the subtitles in the settings could be outside of a `.Section`. This would also apply to subsequent sections that aren't directly under actions.

Maybe `p:first-child`? I still have to test this tho and think of edge cases
